cupyx/jit/_compile.py

```diff
@@ -67,11 +68,16 @@ def transpile(func, attributes, mode, in_types, ret_type):
     if not callable(func):
         raise ValueError('`func` must be a callable object.')
 
-    if func.__name__ == '<lambda>':
-        raise NotImplementedError('Lambda function is not supported.')
-
+    func_name = func.__name__
     attributes = ' '.join(attributes)
     source = jit._getsource_func(func)
+
+    if func_name == '<lambda>':
+        # Program transformation from lambda function to normal function
+        func_name = '_lambda_func'
+        args, source = _preprocess._strip_lambda(source)
+        source = _preprocess._make_function_str(func_name, args, source)
```

Comment:
How about transforming `ast.Lambda` to `ast.FunctionDef` directly, instead of string manipulation?

```py
ast.FunctionDef(func_name, lambda_node.args, [ast.Return(lambda_node.body)], [])
```

Reply:
Fixed in 3af94fc.